specs/Tranche.spec.sol

```diff
@@ -0,0 +1,235 @@
+/*
+ * `num` ranges over non-negative real numbers.
+ */
+
+spec Tranche {
+    Position position; // many tranches can be associated with the same position
+    Token token;
+
+    mapping(address => num) pt; // principal tokens // code.Tranche.balanceOf
+    mapping(address => num) it; // interest tokens // code.InterestToken.balanceOf
+
+    /* until expiration, we have:
+     * - forall x, pt[x] >= it[x]
+     * - sum(it) is equal to the total deposits
+     */
+
+    // deposit underlying tokens
+    function deposit(num amount, address owner) {
+        require(!hasLoss(BUY)); // TODO: it is possible !hasLoss(BUY) && hasLoss(SELL). Is this still safe in that case?
```

Comment:
I see.  Just to clarify:

>Badger has a withdraw fee on its yearn vault fork

Do you (plan to) have a tranche associated to such a yearn vault fork?

> then break amms 

Also, which AMMs do you refer to here?

Reply:
No we aren't going to integrate badger as long as the withdraw fee remains specifically because it breaks this property. Perhaps we should add it as a formal invariant, since I doubt we could make any changes to make it secure. 
On the AMM I just meant that someone dropping the yt price to zero would subject them full loss of their LP deposit.